scout/build/case.py

```diff
@@ -115,52 +115,93 @@ def build_case(case_data, adapter):
     institute_obj = adapter.institute(institute_id)
     if not institute_obj:
         raise IntegrityError("Institute %s not found in database" % institute_id)
+
     case_obj["owner"] = case_data["owner"]
+    case_obj["smn_tsv"] = case_data.get("smn_tsv")
+    case_obj["collaborators"] = get_collaborators(case_data)
+    case_obj["individuals"] = get_individuals(case_data)
+    case_obj["synopsis"] = case_data.get("synopsis", "")
+
+    set_analysis_date(case_obj, case_data)
+    set_assignee(case_obj, case_data)
+    set_causatives(case_obj, case_data)
+    set_suspects(case_obj, case_data)
+    set_timestamps(case_obj)
+
+    case_obj["is_research"] = False
+    case_obj["lims_id"] = case_data.get("lims_id", "")
+    case_obj["rerun_requested"] = False
+    case_obj["research_requested"] = False
+    case_obj["status"] = "inactive"
+    case_obj["panels"] = get_panels(case_data, adapter)
+    case_obj["dynamic_gene_list"] = []
+
+    # Meta data
+    set_genome_build(case_obj, case_data)
+
+    if case_data.get("rank_model_version"):
+        case_obj["rank_model_version"] = str(case_data["rank_model_version"])
 
-    # Owner allways has to be part of collaborators
+    if case_data.get("sv_rank_model_version"):
+        case_obj["sv_rank_model_version"] = str(case_data["sv_rank_model_version"])
+
+    if case_data.get("rank_score_threshold"):
+        case_obj["rank_score_threshold"] = float(case_data["rank_score_threshold"])
+
+    set_cohort(case_obj, case_data)
+    sync_cohort_view(case_obj, institute_obj, adapter)
```

Comment:
Here having 2 functions to set the case cohorts looks like a bit of overkill, especially because the set_cohort one has 2 lines of code, one of which is repeated also sync_cohort_view. For readability I think it would much better to have only one function that sets the case cohorts.

Reply:
Rewritten to one function.